### ISSUE TITLE
Fix ugly JSON (de)serialization hack

### ISSUE THREAD

**Issue by ianprime0509** [revaturelabs/rideforce-user-service#15]
The current method of converting objects with link properties to and from JSON is very hacky and not extensible at all. A much better way would be to change Jackson's (de)serialization behavior directly, eliminating the need for explicit intermediary classes.

Some resources which may be helpful:

- [`BeanPropertyWriter`](https://fasterxml.github.io/jackson-databind/javadoc/2.3.0/com/fasterxml/jackson/databind/ser/BeanPropertyWriter.html)
- [Section 5 of this tutorial](https://www.baeldung.com/jackson-json-view-annotation)

Basically, we should set up a `BeanPropertyWriter` and `BeanPropertyReader` which look for link objects and convert them to or from normal bean objects.

**Comment by ianprime0509** [comment]
Fixed in #19.
